src/main/java/org/folio/circulation/resources/DueDateNotRealTimeScheduledNoticeProcessingResource.java

```diff
@@ -47,12 +47,15 @@ public DueDateNotRealTimeScheduledNoticeProcessingResource(HttpClient client) {
 
   @Override
   protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> findNoticesToSend(
+    ConfigurationRepository configurationRepository,
     ScheduledNoticesRepository scheduledNoticesRepository, PageLimit pageLimit) {
 
-    DateTime timeLimit = LocalDate.now().toDateTime(LocalTime.MIDNIGHT);
-    return scheduledNoticesRepository.findNotices(timeLimit, false,
-      Collections.singletonList(TriggeringEvent.DUE_DATE),
-      FETCH_NOTICES_SORT_CLAUSE, pageLimit);
+    return configurationRepository.findTimeZoneConfiguration()
+      .thenApply(r -> r.map(tz -> ClockManager.getClockManager().getDateTime()
+        .withZone(tz).withTimeAtStartOfDay()))
+      .thenCompose(r -> r.after(timeLimit -> scheduledNoticesRepository.findNotices(timeLimit,
+        false, Collections.singletonList(TriggeringEvent.DUE_DATE),
+        FETCH_NOTICES_SORT_CLAUSE, pageLimit)));
```

Comment:
```suggestion
      .thenCompose(r -> r.after(timeLimit -> findNotices(timeLimit)));
```

Reply:
Do you propose to extract a method here? In this case it'll also require `scheduledNoticesRepository` and `pageLimit` parameters, so it'll look like
```
.thenCompose(r -> r.after(timeLimit -> findNotices(scheduledNoticesRepository, pageLimit,
        timeLimit)));
```
which is probably still cleaner than the original.

Reply:
Sure, I thought those were fixed values.

Reply:
Done